src/percy-agent-client/percy-agent.ts

```diff
@@ -114,4 +127,29 @@ export default class PercyAgent {
 
     return stabilizedDOM.documentElement
   }
+
+  private scopeDomSnapshot(domClone: HTMLElement, scope: string): HTMLElement {
+    try {
+      const elements = Array.from(domClone.querySelectorAll(scope))
+      if (elements.length === 0) {
+        return domClone
+      }
+      // Empty the body element.
+      const body = domClone.querySelector('body')
+      if (!body) {
+        return domClone
+      }
+
+      while (body.firstChild) {
+        body.removeChild(body.firstChild)
+      }
+      // Append only the scoped elements back to the body.
+      elements.forEach((el) => body!.appendChild(el))
+
+      return domClone
+    } catch (error) {
+      console.log(`[percy] Could not scope snapshot to: ${scope} ; Error: ${error}`)
```

Comment:
Nice graceful handling here. I think that makes sense. If a bad selector is sent through, falling back to no scope behavior makes sense to me